### PR TITLE
Full-code

### PR DESCRIPTION
**Pull request by banzaman**
- Use ruby gems as software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Use params from browser request in a safe way.
- Use preprocessed html file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a webapp that requires the user to log in.
- Use devise gem for authentication.

### CODE REVIEW

**Comment by shahadat3669** [comment]
Hi @banzaman, I noticed that after requesting a review, you pushed a new commit. This is not a good habit as it makes the review process difficult. If you make new changes during the review process, it will be harder to review them. My recommendation is to wait until you receive feedback before pushing any new changes.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## Live Demo <a name="live-demo"></a>
+
+Visit the site live [here.](https://mysite-28qd.onrender.com)
+
+You can see the video presentation for this project [here.](https://www.loom.com/share/8f12e2777cdc4a198036debc24bcd1)
+
```
Comment (Efiamotu-1) [inline README.md:55]: - [ ] The links you have provided are not functioning correctly as shown in the images below please kindly work on this :+1: 
vid
vi

File: app/controllers/transactions_controller.rb
Comment (Efiamotu-1) [inline app/controllers/transactions_controller.rb:1]: - [ ] From the project design below, the buttons for adding new category and transaction should be fixed to the bottom of the screen, right now this is not the case please kindly do his
- project design
bt
- your design
btt

Comment (Efiamotu-1) [inline app/controllers/transactions_controller.rb:1]: - [ ] For the add new transactions page, you are missing the option for adding a category as shown in the image below, please kindly add this
adn

Comment (Efiamotu-1) [inline app/controllers/transactions_controller.rb:1]: - [ ] Your tests are also failing, please kindly fix them :+1: 
ft

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @banzaman , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by OyePriscilla** [comment]
Hi @banzaman 👍 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that has wrong flow and was not approved last week. Kindly open a new PR and ensure that you use gitflow. 🙏 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**